src/inputs/input-with-icon.scss

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import './input';
+
+// Mixin to have inputs combined with icon (eg. select)
+@mixin iui-input-with-icon {
+  @include iui-input;
+  display: grid;
+  align-items: center;
+  grid-template: 'input icon' / 1fr auto;
+  margin: 0;
+  padding: 0;
+  vertical-align: baseline;
+  position: relative;
+  width: 100%;
+  @media (prefers-reduced-motion: no-preference) {
+    transition: border-color $iui-speed-fast ease-out;
+  }
+
+  &:hover {
+    @include themed {
+      border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
+    }
+
+    &::after {
+      border-bottom: none;
+      @include themed {
+        border-top: $iui-after-border t(iui-icons-color-actionable-hover);
+      }
+    }
+  }
+
+  &.iui-active {
+    &::after {
+      border-top: none;
+      @include themed {
+        border-bottom: $iui-after-border t(iui-icons-color-actionable);
+      }
+    }
+
+    &:hover::after {
+      @include themed {
+        border-bottom: $iui-after-border t(iui-icons-color-actionable-hover);
+      }
+    }
+  }
+
+  &.iui-disabled {
+    @include themed {
+      background-color: t(iui-color-background-disabled);
+      border-color: t(iui-color-background-disabled);
+    }
+
+    .iui-end-icon {
```

Comment:
It looks like we are not really changing any styles on this icon when it's disabled. So if you just use JS to remove the `iui-actionable` class when disabled, then this CSS will not be needed. See `ComboBox.tsx:316` for how that looks in React.

Reply:
removed iui-actionable on disabled icons

Reply:
I see you've added back `iui-disabled` on the icon for the cursor. It can be removed if you do this instead:
1. apply `pointer-events: none` on the base icon itself
2. apply `pointer-events: auto` inside `iui-actionable` modifier

Reply:
moved disabled to `iui-input-with-icon` because end-icon needs disabled colors. Also added pointer-events: none there for disabled combobox.